Formula/vile.rb

```diff
@@ -0,0 +1,31 @@
+class Vile < Formula
+  desc "Vi Like Emacs Editor"
+  homepage "https://invisible-island.net/vile/"
+  url "https://invisible-island.net/archives/vile/current/vile-9.8v.tgz"
+  sha256 "240edec7bbf3d9df48b3042754bf9854d9a233d371d50bba236ec0edd708eed5"
+  license "GPL-2.0-or-later"
+
+  depends_on "expect" => :test
+
+  uses_from_macos "flex" => :build
+  uses_from_macos "groff" => :build
+  uses_from_macos "libiconv"
+  uses_from_macos "ncurses"
+  uses_from_macos "perl"
+
+  def install
+    system "./configure", *std_configure_args,
+                          "--disable-imake",
+                          "--enable-colored-menus",
+                          "--with-ncurses",
+                          "--with-screen=ncurses",
+                          "--without-x"
+    system "make", "install"
+  end
+
+  test do
+    pipe_output("env TERM=xterm expect -",
+      "spawn vile;expect \"unnamed\";send \":w new\r:q\r\";expect eof")
```

Comment:
Can we use Ruby's `PTY` module instead of `expect`?

Reply:
I'm not sure...  I've never touched Ruby before 48 hours ago!  (I'm a C and assembly person, so I'll defer to the judgement of the committers here.)

Reply:
We probably can -- `grep -r 'require "pty"` for many examples.